test/fortran/f_c_string/Test_f_c_string_t_mod.f90

```diff
@@ -0,0 +1,39 @@
+module Test_f_c_string_t_mod
+    use fortran_test_framework_mod, only : assertEqual, assert
+    use f_c_string_t_mod, only : f_c_string_t
+    use f_c_string_i_mod, only : strlen
+    use Test_f_c_string_i_mod, only : to_upper
+    use iso_c_binding, only : c_null_char, c_ptr
+    implicit none
+    contains
+    subroutine Test_f_c_string_t()
```

Comment:
It might be worth adding a test for empty / zero-length strings. Generally, we should try to cover all corner cases with these tests in my opinion.